cpp/arcticdb/version/version_utils.hpp

```diff
@@ -217,12 +217,22 @@ inline bool load_latest_ongoing(const LoadParameter& load_params, const std::sha
     return false;
 }
 
-inline bool looking_for_undeleted(const LoadParameter& load_params, const std::shared_ptr<VersionMapEntry>& entry) {
-    if(!(load_params.load_type_ == LoadType::LOAD_UNDELETED && entry->tombstone_all_))
+inline bool looking_for_undeleted(const LoadParameter& load_params, const std::shared_ptr<VersionMapEntry>& entry, const VersionId& oldest_loaded_index) {
+    if(load_params.load_type_ != LoadType::LOAD_UNDELETED) {
         return true;
+    }
 
-    ARCTICDB_DEBUG(log::version(), "Exiting because we have found an undeleted version");
-    return false;
+    if(entry->tombstone_all_) {
+        const bool is_deleted_by_tombstone_all = entry->tombstone_all_->version_id() >= oldest_loaded_index;
+        if(is_deleted_by_tombstone_all) {
+            ARCTICDB_DEBUG(log::version(), "Exiting because we have found an undeleted version");
```

Comment:
I might add in the version that we found, as it will make it easier to debug

Reply:
Done